viam-cartographer/src/carto_facade/carto_facade_test.cc

```diff
@@ -1440,73 +1575,187 @@ BOOST_AUTO_TEST_CASE(CartoFacade_demo_with_imu) {
+        BOOST_TEST(viam_carto_get_position(vc, &pr) == VIAM_CARTO_SUCCESS);
+        BOOST_TEST(pr.x != 0);
+        BOOST_TEST(pr.y != 0);
+        BOOST_TEST(pr.z == 0);
+        BOOST_TEST(pr.imag != 0);
+        BOOST_TEST(pr.jmag != 0);
+        BOOST_TEST(pr.kmag != 0);
+        BOOST_TEST(pr.real != 1);
+        BOOST_TEST(to_std_string(pr.component_reference) == "lidar");
+
+        BOOST_TEST(viam_carto_get_position_response_destroy(&pr) ==
                    VIAM_CARTO_SUCCESS);
-        double readings[6] = {0.2, 0, 9.8, -0.6, 0, 0};
-        add_new_testIMUReading(vc, readings, 1629037855000200);
     }
 
-    // GetPosition changed from 3 successful AddLidarReading requests
+    // GetPointCloudMap after 4 successful sensor readings
+    {
+        viam_carto_get_point_cloud_map_response mr;
+        BOOST_TEST(viam_carto_get_point_cloud_map(vc, &mr) ==
+                   VIAM_CARTO_SUCCESS);
+        pcl::PCLPointCloud2 blob;
+        pcl::PointCloud<pcl::PointXYZRGB>::Ptr cloud(
+            new pcl::PointCloud<pcl::PointXYZRGB>);
+        auto s = to_std_string(mr.point_cloud_pcd);
+        BOOST_TEST(viam::carto_facade::util::read_pcd(s, blob) == 0);
+        pcl::fromPCLPointCloud2(blob, *cloud);
+        BOOST_TEST(cloud != nullptr);
+        BOOST_TEST(cloud->points.size() != 0);
+        BOOST_TEST(viam_carto_get_point_cloud_map_response_destroy(&mr) ==
+                   VIAM_CARTO_SUCCESS);
+    }
+
+    // GetInternalState after 4 successful sensor readings
+    {
+        viam_carto_get_internal_state_response isr;
+        BOOST_TEST(viam_carto_get_internal_state(vc, &isr) ==
+                   VIAM_CARTO_SUCCESS);
+        // on arm64 linux this is strictly greater than;
+        // on arm64 mac for some reason it is equal to;
+        // https://viam.atlassian.net/browse/RSDK-3866
+        BOOST_TEST(blength(isr.internal_state) >=
+                   last_internal_state_response_size);
+        last_internal_state_response_size = blength(isr.internal_state);
+        BOOST_TEST(viam_carto_get_internal_state_response_destroy(&isr) ==
+                   VIAM_CARTO_SUCCESS);
+    }
+
+    // add fifth sensor reading: only odometer
+    {
+        double odometer_reading[7] = {0.5, 0.2, 9.8, 0.6, 0.1, 0, 0.8};
+        add_odometer_reading_successfully(vc, 3, odometer_reading,
+                                          1629037857000200);
+    }
+
+    // GetPosition changed from 3 successful AddLidarReading, 4 successful
+    // AddIMUReading, and 1 successful AddOdometerReading requests
     {
         viam_carto_get_position_response pr;
         BOOST_TEST(viam_carto_get_position(vc, &pr) == VIAM_CARTO_SUCCESS);
-        BOOST_TEST(pr.x == 0);
-        BOOST_TEST(pr.y == 0);
+        BOOST_TEST(pr.x != 0);
+        BOOST_TEST(pr.y != 0);
         BOOST_TEST(pr.z == 0);
-        BOOST_TEST(pr.imag == 0);
-        BOOST_TEST(pr.jmag == 0);
-        BOOST_TEST(pr.kmag == 0);
-        BOOST_TEST(pr.real == 1);
```

Comment:
There was something wrong before since those values were supposed to have been non-zero. This might be connected to the fact that the IMU data had the same timestamps as the lidar data. Either way, this is fixed now with this PR.